lib/spack/spack/util/web.py

```diff
@@ -66,6 +62,10 @@ def http_error_default(self, req, fp, code, msg, hdrs):
 
 
 def _urlopen():
+    import spack.config
```

Comment:
if you have to add inline imports like this on almost everywhere, it's better not to do it at all

Reply:
> I don't think this is an improvement, rather the opposite :( 

I'll try to break circular deps, but I have to say this is _exactly_ what we have on develop. It's just that using processes with spawn context, instead of threads, makes the circular deps appearent.

Reply:
For the record, current cycles on `develop`:
![spack_util_web](https://github.com/spack/spack/assets/4199709/140e74da-01c9-4aa1-a8c3-228a031d7b34)